python/ql/lib/semmle/python/frameworks/Ldap.qll

```diff
@@ -0,0 +1,80 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `python-ldap` PyPI package (imported as `ldap`).
+ * See https://www.python-ldap.org/en/python-ldap-3.3.0/index.html
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+
+/**
+ * Provides models for the `python-ldap` PyPI package (imported as `ldap`).
+ *
+ * See https://www.python-ldap.org/en/python-ldap-3.3.0/index.html
+ */
+private module Ldap {
+  /**
+   * The name of an `ldap` method used to execute a query.
+   *
+   * See https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#functions
+   */
+  private string ldapQueryMethodName() {
+    result in ["search", "search_s", "search_st", "search_ext", "search_ext_s"]
+  }
+
+  /** The execution of an `ldap` query. */
```

Comment:
No reason to keep this in a predicate, since it is only used once

```suggestion
  /**
   * The execution of an `ldap` query.
   *
   * See https://www.python-ldap.org/en/python-ldap-3.3.0/reference/ldap.html#functions
   */
```

Reply:
Agreed.